middleman-core/lib/middleman-core/sitemap/extensions/traversal.rb

```diff
@@ -19,14 +19,26 @@ def parent
           parts = root.split('/')
 
           tail = parts.pop
-          is_index = (tail == @app.config[:index_file])
+
+          is_index = if @app.config[:traversal_use_any_index]
+            tail.start_with?("index.")
+          else
+            tail == @app.config[:index_file]
+          end
 
           if parts.empty?
-            return is_index ? nil : @store.find_resource_by_path(@app.config[:index_file])
+            return nil if is_index 
+
+             if @app.config[:traversal_use_any_index]
+              return @store.find_index_resources_by_prefix("").first
+            else
+              return @store.find_resource_by_path(@app.config[:index_file])
+            end
```

Comment:
I believe the `find_index_resources_by_prefix` function should return the appropriate resources even when `:traversal_use_any_index` is configured to false. This would allow merely calling it here rather than testing that configuration option and falling back to some other behavior when that configuration is false.

More importantly, it would prevent calling the `find_index_resources_by_prefix` later, only to find out that it doesn’t return index resources.

Reply:
Trying to avoid possible regressions with large existing sites with many index files. Need to be very defensive when there are decade-old projects still running. Can make this the default in the future if all goes well.